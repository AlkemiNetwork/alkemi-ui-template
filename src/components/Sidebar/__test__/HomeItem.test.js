import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

import HomeIcon from "@material-ui/icons/Home";

import ListItem1 from "../../UI/ListItem/1";
import SidebarHomeItem from "../HomeItem";

describe("<SidebarHomeItem />", () => {
    let wrapper;
    let props = {};

    beforeEach(() => {
        props = {
            onClick: jest.fn()
        };
        wrapper = shallow(<SidebarHomeItem {...props} />);
    });

    it("renders 1 ListItem1", () => {
        expect(wrapper.find(ListItem1).length).toEqual(1);
    });

    it("renders 1 HomeIcon", () => {
        expect(
            wrapper
                .dive()
                .dive()
                .find(HomeIcon).length
        ).toEqual(1);
    });

    it("ListItem1 button is functional", () => {
        wrapper
            .find(ListItem1)
            .at(0)
            .simulate("click");
        expect(props.onClick.mock.calls.length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
