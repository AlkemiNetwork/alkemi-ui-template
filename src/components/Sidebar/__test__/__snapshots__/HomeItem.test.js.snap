// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarHomeItem /> matches snapshot 1`] = `
<WithStyles(ListItem1)
  icon={<pure(HomeIcon) />}
  onClick={[MockFunction]}
  text="Home"
/>
`;
